File: app/src/main/java/com/m0n0l0c0/LinkedChronos/ui/ChronoAdapter.java
package com.m0n0l0c0.LinkedChronos.ui;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.m0n0l0c0.LinkedChronos.R;
import com.m0n0l0c0.LinkedChronos.interfaces.CheckEmpty;
import com.m0n0l0c0.LinkedChronos.model.Chrono;

import java.util.ArrayList;
import java.util.concurrent.TimeUnit;

/**
 * Created by juanje on 22/05/16.
 */
public class ChronoAdapter extends RecyclerView.Adapter<ChronoAdapter.ChronoViewHolder> {

    private ArrayList<Chrono> data;
    private Context context;
    private static final String FORMAT = "%02d:%02d:%02d";

    public static class ChronoViewHolder extends RecyclerView.ViewHolder {

        private TextView name;
        private TextView time;

        public ChronoViewHolder(View itemView) {
            super(itemView);

            name = (TextView)itemView.findViewById(R.id.chrono_name);
            time = (TextView)itemView.findViewById(R.id.chrono_time);

        }

        public void bindChrono(Chrono chrono){
            name.setText(chrono.getName());

            long milis = chrono.getMiliseconds();

            time.setText(String.format(FORMAT, TimeUnit.MILLISECONDS.toHours(milis),
                    TimeUnit.MILLISECONDS.toMinutes(milis) - TimeUnit.HOURS.toMinutes(
                            TimeUnit.MILLISECONDS.toHours(milis)),
                    TimeUnit.MILLISECONDS.toSeconds(milis) - TimeUnit.MINUTES.toSeconds(
                            TimeUnit.MILLISECONDS.toMinutes(milis))));
        }
    }

    public ChronoAdapter(ArrayList<Chrono> data, Context context) {
        this.data = data;
        this.context = context;
    }

    @Override
    public ChronoViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {

        View itemView = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.list_item, viewGroup, false);

        ImageView delete = (ImageView) itemView.findViewById(R.id.delete_chrono);
        final int position = i;
        delete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                deleteData(position);
                ((CheckEmpty)context).checkEmptyView();
            }
        });

        ChronoViewHolder chronoViewHolder = new ChronoViewHolder(itemView);

        return chronoViewHolder;
    }

    @Override
    public void onBindViewHolder(ChronoViewHolder chronoViewHolder, int i) {
        Chrono item = data.get(i);
        chronoViewHolder.bindChrono(item);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void insertData(int pos, Chrono chrono){
        data.add(pos, chrono);
        notifyItemInserted(pos);
    }

    public void deleteData(int pos){
        data.remove(pos);
        notifyItemRemoved(pos);
    }

}

File: app/src/main/java/com/m0n0l0c0/linkedchronos/ui/home/ChronoAdapter.kt
package com.m0n0l0c0.linkedchronos.ui.home

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.m0n0l0c0.linkedchronos.R
import com.m0n0l0c0.linkedchronos.interfaces.CheckEmpty
import com.m0n0l0c0.linkedchronos.model.Chrono
import com.m0n0l0c0.linkedchronos.ui.home.ChronoAdapter.ChronoViewHolder
import java.util.*
import java.util.concurrent.TimeUnit

/**
 * Created by juanje on 22/05/16.
 */
class ChronoAdapter(private val data: ArrayList<Chrono>, private val context: Context) : RecyclerView.Adapter<ChronoViewHolder>() {

    class ChronoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val name: TextView = itemView.findViewById(R.id.chrono_name)
        private val time: TextView = itemView.findViewById(R.id.chrono_time)

        fun bindChrono(chrono: Chrono) {
            name.text = chrono.name

            val millis = chrono.miliseconds

            time.text = String.format(FORMAT, TimeUnit.MILLISECONDS.toHours(millis),
                    TimeUnit.MILLISECONDS.toMinutes(millis) - TimeUnit.HOURS.toMinutes(
                            TimeUnit.MILLISECONDS.toHours(millis)),
                    TimeUnit.MILLISECONDS.toSeconds(millis) - TimeUnit.MINUTES.toSeconds(
                            TimeUnit.MILLISECONDS.toMinutes(millis)))
        }
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ChronoViewHolder {

        val itemView = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.list_item, viewGroup, false)

        val delete = itemView.findViewById<ImageView>(R.id.delete_chrono)
        delete.setOnClickListener {
            deleteData(i)
            (context as CheckEmpty).checkEmptyView()
        }

        return ChronoViewHolder(itemView)
    }

    override fun onBindViewHolder(chronoViewHolder: ChronoViewHolder, i: Int) {
        val item = data[i]
        chronoViewHolder.bindChrono(item)
    }

    override fun getItemCount(): Int {
        return data.size
    }

    fun insertData(pos: Int, chrono: Chrono) {
        data.add(pos, chrono)
        notifyItemInserted(pos)
    }

    private fun deleteData(pos: Int) {
        data.removeAt(pos)
        notifyItemRemoved(pos)
    }

    companion object {
        private const val FORMAT = "%02d:%02d:%02d"
    }

}
